File: springboot_app/src/main/java/com/hygorp/biblioteca/service/GenreService.java
package com.hygorp.biblioteca.service;

import com.hygorp.biblioteca.domain.Author;
import com.hygorp.biblioteca.domain.Book;
import com.hygorp.biblioteca.domain.Genre;
import com.hygorp.biblioteca.repository.GenreRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class GenreService {

    private final GenreRepository genreRepository;

    private GenreService(GenreRepository genreRepository){
        this.genreRepository = genreRepository;
    }

    public List<Genre> findAllGenres(){
        return genreRepository.findAll();
    }

    public Genre findGenreById(Integer id){
        Optional<Genre> genreObj = genreRepository.findById(id);
        return genreObj.orElseThrow(()-> new RuntimeException("Genre not found"));
    }

    public List<Genre> findGenresByName(String name){
        return genreRepository.findAllByName(name);
    }

    public Genre insert(Genre obj){
        return genreRepository.save(obj);
    }

    public Genre update(Genre obj){
        Genre genreObj = findGenreById(obj.getId());
        genreObj.setName(obj.getName());
        genreObj.getBooks().addAll(obj.getBooks());
        return genreRepository.save(genreObj);
    }

    public void delete(Integer id){
        genreRepository.deleteById(id);
    }
}

File: vuejs_frontend/src/services/GenreService.js
import http from "../http-common"

class GenreService {
    findAll(){
        return http.get("/genres");
    }

    findById(id) {
        return http.get(`/genres/${id}`)
    }

    create(data) {
        return http.post("/genres", data);
    }

    update(id, name) {
        return http.put(`/genres/${id}`, name);
    }

    delete(id) {
        return http.delete(`/genres/${id}`)
    }
}

export default new GenreService();